fastqpreprocessing/src/alignment_datatype.cpp

```diff
@@ -123,24 +140,75 @@ void LineFields::writeTabbedToFile(std::ofstream& outfile)
           << molecule_barcode_base_above_30 << "\n";
 }
 
-std::string LineFields::getNextField(std::string const& s)
+void LineFieldsParser::crashLF(std::string msg)
+{
+  crash(msg + "\nThe bad line:\n" + s_);
+}
+
+int LineFieldsParser::getNextFieldInt()
```

Comment:
This block here is all actual code change, to fix bugs / better handle corner cases I found through testing. I could split these changes into another PR, if you'd prefer.